services/QuillLMS/app/serializers/user_admin_serializer.rb

```diff
@@ -3,7 +3,7 @@ class UserAdminSerializer < ActiveModel::Serializer
 
   def teachers
     teacher_ids = User.find(object.id).admins_teachers
-    if teacher_ids.any?
+    if teacher_ids&.any?
```

Comment:
What about moving the check for empty_ids inside the `TeachersData.run` method itself:

```
  def self.run(teacher_ids)
   return [] if teacher_ids.blank?

    teacher_ids_str = teacher_ids.join(', ')
    ...
```
That way, the `TeachersData.run` is now protected from other objects calling it with nil/empty params.

Reply:
Makes sense. I moved the whole check into the query method.

Reply:
Perfect!